basic_module/distribution/src/core.rs

```diff
@@ -0,0 +1,7 @@
+pub trait Distribution {
```

Comment:
Distribution module should record each validator's contribution when abci `BeginBlock` and distribute should be called when abci "EndBlock" if the module-defined term ends. `BeginBlockHandler` and `EndBlockHandler` can be more clear for those trait names.

Reply:
I prefer `open_block` and `close_block`. We use the names.

Reply:
How about these trait names `OpenBlockHandler` and `CloseBlockHandler`?